dask/local.py

```diff
@@ -503,6 +505,13 @@ def fire_task():
     return nested_get(result, state["cache"])
 
 
+def executor_apply_async(executor, func, args=(), kwds={}, callback=None):
+    """ A apply_async implementation for `concurrent.futures.Executor`s """
+    fut = executor.submit(func, *args, **kwds)
+    if callback is not None:
+        fut.add_done_callback(lambda f: callback(f.result()))
```

Comment:
Any thoughts on changing the scheduler itself to use `submit` and `result` other than `apply_async`?

Reply:
Yeah I think that would be the next step. Was trying to initially see how doable it would be to just switch to `Executor`s. Plus it's helping build confidence being able to run things with these `Executor`s and run the test suite.

I think the ideal case is we just collect the `Future`s and add them into the Dask graph for operations like `.persist()`. Only calling `.result()` when the user calls `.compute()` and we need to materialize things.

Reply:
At this point I don't think that we're going to inject anything into the dask graph.  Also, `persist` doesn't really do anything different for this scheduler.  

My hope is that you can find-replace `s/apply_async/submit/` and `s/get/result`, clean things up, and then you'll be done.

Reply:
So do you think that is not achievable or just not done today?

Reply:
I'm sorry, I guess I don't understand the objective here clearly.  I think that the objective for this PR is to see if we can replace apply_async/get with submit/result.  I don't see a need to modify the task graph in order to acheive this result.  I may not understand your intention though.

Reply:
I don't think that needs to happen in this PR. Though it is a motivating use case for me. So I'm curious to know whether you think that is achievable.

Reply:
For context I'm thinking about this thread ( https://github.com/dask/dask/issues/3235 ) and in particular this comment ( https://github.com/dask/dask/issues/3235#issuecomment-369737875 ).

Reply:
Have gotten rid of `apply_async`. It now takes an `Executor` object and calls `Executor.submit` internally.